lib/mix/lib/mix/tasks/compile.xref.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Mix.Tasks.Compile.Xref do
+  use Mix.Task
+  alias Mix.Tasks.Compile.Elixir, as: E
+
+  @recursive true
+  @manifest ".compile.xref"
+  @manifest_vsn :v1
+
+  @moduledoc """
+  Performs remote dispatch checking.
+
+  When this task runs, it will check the modification time of the `:elixir`
+  compiler manifest. If it has changed, `mix xref` will be run to check remote
+  dispatches. You can force checking regardless of modification time by passing
+  the `--force` option.
+
+  ## Command line options
+
+    * `--force` - forces checking regardless of modification time
+    * `--warnings-as-errors` - treat warnings as errors and return a non-zero exit code
+
+  """
+
+  @doc """
+  Runs this task.
+  """
+  @spec run(OptionParser.argv) :: :ok | :noop
+  def run(args) do
+    {opts, _, _} =
+      OptionParser.parse(args, switches: [force: :boolean, warnings_as_errors: :boolean])
+
+    if Mix.Utils.stale?(E.manifests(), manifests()) or opts[:force] do
+      if Mix.Task.run("xref", args) == :ok do
+        write_manifest()
```

Comment:
You should always write this manifest if you ran the task and if it did not fail.